main/Telemetry.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * @file Telemetry.cpp
+ * @author your name (you@domain.com)
+ * @brief 
+ * @version 0.1
+ * @date 2024-11-01
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef TELEMETRY_CPP
+#define TELEMETRY_CPP
+
+#include <initializer_list>
+#include <cstring>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+
+#include "driver/uart.h"
+
+#include "esp_mac.h"
+
+#include "globalVar.h"
+#include "taskClass.hpp"
+
+template<int n_channels>
+class Telemetry : public Task {
```

Comment:
Interesting class. This class should be think in future to be more generic and maybe implement a wifi telementry class. Think it for your final project